boot/assets.go

```diff
@@ -355,7 +355,16 @@ func TrustedAssetsUpdateObserverForModel(model *asserts.Model, gadgetDir string)
 	}
 	// trusted assets need tracking only when the system is using encryption
 	// for its data partitions
-	trackTrustedAssets := hasSealedKeys(dirs.GlobalRootDir)
+	trackTrustedAssets := false
+	_, err := sealedKeysMethod(dirs.GlobalRootDir)
+	switch {
+	case err == nil:
+		trackTrustedAssets = true
+	case err == errSealingNoKeys:
+		// nothing
+	case err != nil && err != errSealingNoKeys:
```

Comment:
minor nitpick, but this is logically impossible to have err == nil here, since we explicitly switch on that above, so I think it's slightly more readable as:

```suggestion
	case err != nil:
	// all other errors
```